lib/ansible/modules/web_infrastructure/rundeck_project.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+"""
+Ansible module to manage rundeck projects
+(c) 2017, Loic Blot <loic.blot@unix-experience.fr>
+Sponsored by Infopro Digital. http://www.infopro-digital.com/
+Sponsored by E.T.A.I. http://www.etai.fr/
+
+This file is part of Ansible
+
+Ansible is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Ansible is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: rundeck_project
+
+short_description: Manage Rundeck projects.
+description:
+    - Create and remove Rundeck projects through HTTP API.
+version_added: "2.4"
+author: "Loic Blot (@nerzhul)"
+options:
+    state:
+        description:
+            - Create or remove Rundeck project.
+        choices: ['present', 'absent']
+        default: 'present'
+    name:
+        description:
+            - Sets the project name.
+        required: True
+    url:
+        description:
+            - Sets the rundeck instance URL.
+        required: True
+    api_version:
+        description:
+            - Sets the API version used by module.
+            - API version must be at least 14.
+        default: 14
+    token:
+        description:
+            - Sets the token to authenticate against Rundeck API.
+        required: True
+
+
+'''
+EXAMPLES = '''
+- name: Create a rundeck project
+  rundeck_project:
+    name: "Project_01"
+    api_version: 18
+    url: "https://rundeck.example.org"
+    token: "mytoken"
+    state: present
+
+- name: Remove a rundeck project
+  rundeck_project:
+    name: "Project_02"
+    url: "https://rundeck.example.org"
+    token: "mytoken"
+    state: absent
+'''
+
+RETURN = '''
+rundeck_response:
+    description: Rundeck response when a failure occurs
+    returned: failed
+    type: string
+before:
+    description: dictionnary containing project informations before modification
+    returned: success
+    type: dictionnary
+after:
+    description: dictionnary containing project informations after modification
+    returned: success
+    type: dictionnary
+'''
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+import json
+
+
+class RundeckProjectManager:
```

Comment:
Good idea to make all classes new-style objects.  You can do that by inheriting from object